File: 121.Best Time to Buy and Sell Stock/solution.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        maxprofit = 0
        buy, sell = 0,1
        n = len(prices)

        while sell<n:
            if prices[sell] > prices[buy]:
                profit = prices[sell] - prices[buy]
                if maxprofit < profit:
                    maxprofit = profit
            else:
                buy = sell
            sell += 1
        return maxprofit
        

File: 121.Best Time to Buy and Sell Stock/solution.java
class Solution {
    public int maxProfit(int[] prices) {
        int buy = 0, sell = 1;
        int maxprofit = 0;
        int n = prices.length;

        while(sell<n){
            if(prices[sell]>prices[buy]){
                maxprofit = Math.max(maxprofit,prices[sell]-prices[buy]);
            }else{
                buy = sell;
            }
            sell++;
        }
        return maxprofit;
    }
}